tests/components/radiation/test_radiation.py

```diff
@@ -3,77 +3,201 @@
-    assert rad.var_units("radiation__ratio_to_flat_surface") == "None"
+def test_turbidity():
+    grid = RasterModelGrid((5, 5), xy_spacing=10e0)
+    grid.add_zeros("topographic__elevation", at="node")
 
+    radiations = []
+    for value in np.linspace(0.0, 1.0):
+        radiation = Radiation(
+            grid, latitude=0.0, clearsky_turbidity=value, opt_airmass=1.0
+        )
+        radiation.update()
+        radiations.append(grid.at_cell["radiation__net_flux"].max())
 
-def test_grid_shape(rad):
-    assert rad.grid.number_of_node_rows == _SHAPE[0]
-    assert rad.grid.number_of_node_columns == _SHAPE[1]
+    assert np.all(np.diff(radiations) < 0.0)
 
 
-def test_grid_x_extent(rad):
-    assert rad.grid.extent[1] == (_SHAPE[1] - 1) * _SPACING[1]
+@pytest.mark.parametrize("time_0,time_1", [(0.0, 1.0), (2000.5, 2010.5), (-0.5, 0.5)])
+def test_time(time_0, time_1):
+    grid = RasterModelGrid((5, 5), xy_spacing=10e0)
+    grid.add_zeros("topographic__elevation", at="node")
+    radiation = Radiation(grid, latitude=0.0, current_time=time_0)
+    radiation.update()
 
+    radiation_at_time_0 = grid.at_cell["radiation__net_shortwave_flux"].copy()
 
-def test_grid_y_extent(rad):
-    assert rad.grid.extent[0] == (_SHAPE[0] - 1) * _SPACING[0]
+    radiation = Radiation(grid, latitude=0.0, current_time=time_1)
+    radiation.update()
 
+    radiation_at_time_1 = grid.at_cell["radiation__net_shortwave_flux"]
 
-def test_field_getters(rad):
-    for name in rad.grid["node"]:
-        field = rad.grid["node"][name]
-        assert isinstance(field, np.ndarray)
-        assert field.shape == (
-            rad.grid.number_of_node_rows * rad.grid.number_of_node_columns,
-        )
+    assert np.allclose(radiation_at_time_0, radiation_at_time_1)
 
-    for name in rad.grid["cell"]:
-        field = rad.grid["cell"][name]
-        assert isinstance(field, np.ndarray)
-        assert field.shape == (
-            rad.grid.number_of_cell_rows * rad.grid.number_of_cell_columns,
-        )
 
-    with pytest.raises(KeyError):
-        rad.grid["not_a_var_name"]
+TIMES = {
+    "march_equinox": 1.39 * 365.0 / (2.0 * np.pi),
+    "june_solstice": (np.pi / 2.0 + 1.39) * 365 / (2.0 * np.pi),
+    "september_equinox": (np.pi + 1.39) * 365.0 / (2.0 * np.pi),
+    "december_solstice": (3.0 * np.pi / 2.0 + 1.39) * 365.0 / (2.0 * np.pi),
+}
+
+
+@pytest.mark.parametrize(
+    "latitude,solstice",
+    [
+        (70.0, "december"),
+        (80.0, "december"),
+        (-70.0, "june"),
+        (-80.0, "june"),
+    ],
+)
+def test_solstice(solstice, latitude):
+    time = TIMES[f"{solstice}_solstice"] / 365.0
+
+    grid = RasterModelGrid((5, 5), xy_spacing=10e0)
+    grid.add_zeros("topographic__elevation", at="node")
+    radiation = Radiation(grid, latitude=latitude, current_time=time)
+    radiation.update()
+
+    assert np.all(np.isnan(grid.at_cell["radiation__extraterrestrial_flux"]))
```

Comment:
What do you think about this? It seems like for cases where the sun doesn't rise, the radiation should be 0.0, not ``nan``. I added this assertion just because that's the way it is now, but it seems like it should be,
```python
assert np.allclose(grid.at_cell["radiation__extraterrestrial_flux"], 0.0)
```

Reply:
Added code to force any cases of NaN values to 0.0, added the allclose(0.0) assertion you mentioned above, this is resolved.